VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+            $path = Join-Path $pwd ""
+            $operationTypeJson = '{
+                "operationType": "INSTALL",
+                '
+              $resourcesBodyObject += [pscustomobject]@{
+                  resources = $resourcesObject
+              }
+              $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $path"temp.json"
+              Get-Content $path"temp.json" | Select-Object -Skip 1 | Set-Content $path"temp1.json"
+              $resouresJson = Get-Content $path"temp1.json" -Raw
+              Remove-Item -Path $path"temp.json"
+              Remove-Item -Path $path"temp1.json"
+              $requestCertificateSpecJson = $operationTypeJson + $resouresJson
+              $requestCertificateSpecJson | Out-File $path"$($workloadDomain)-updateCertificateSpec.json"
+
+              # Install Certificate
+              Try {
+                Write-Output "Installing Signed Certificates for Components Associated with Domain $($workloadDomain) (can take 60+ mins)"
+                $myTaskId = Set-VCFCertificate -domainName $($workloadDomain) -json $path"$($workloadDomain)-updateCertificateSpec.json"
+                $pollLoopCounter = 0
+                Do {
+                    If ($pollLoopCounter % 10 -eq 0) {
+                        Write-Output "Checking Installation of Signed Certificates for Components Associated with Domain $($workloadDomain)"
+                    }
+                    $response = Get-VCFTask $myTaskId.id
+                    If ($response.status -in "In Progress","IN_PROGRESS") {
+                        If (($pollLoopCounter % 10 -eq 0) -AND ($pollLoopCounter -gt 9)) {
+                            Write-Output "Certificate installation still in progress"
+                        }
+                        Start-Sleep 60
+                        $pollLoopCounter ++
+                    }
+                } While ($response.status -in "In Progress","IN_PROGRESS")
+                If ($response.status -eq "FAILED") {
+                    Write-Error "Workflow Completed with status $($response.status)" 
+                }
+                elseIf ($response.status -eq "SUCCESSFUL") {
+                    Write-Output "Workflow Completed with status $($response.status)"
+                }
+                else {
+                    Write-Output "Workflow Completed with an unrecognized status of $($response.status). Please check before proceeding."
+                }
+                Write-Output "Signed Certificates for Components Associated with Domain $($workloadDomain) Completed with Status: $($response.status)"
+                Write-Output "Finished the Process of Installing Signed Certificates to Components Managed by SDDC Manager"
+            }
+            Catch {
+                $ErrorMessage = $_.Exception.Message
+                Write-Error "Error was: $ErrorMessage"
+            }
+        } else {
+            Write-Error "cannot authenticate to SDDC Manager"
```

Comment:
```suggestion
            Write-Error "Unable to authenticate to SDDC Manager ($($server)): PRE_VALIDATION_FAILED"
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)